README.md

```diff
@@ -1,3 +1,262 @@
+minifier.renameProgram(['path/to/file.ts', 'path/to/another/file.ts', 'some/other/file.d.ts'], 'path/to/destination');
+```
+
+Transpile the above script into JavaScript using the TypeScript compiler and run it with Node:
+
+```shell
+tsc --module commonjs script.ts
+node script.js
+```
+
+A new instance of the minifier takes an optional constructor argument:
+
+```javascript
+MinifierOptions {
+  failFast?: boolean;
+  basePath?: string;
+}
+```
+
+`failFast`: Setting `failFast` to true will throw an error when the minifier hits one.
+
+`basePath`:  Specifies the base path that the minifier uses to figure out to where to write the minfied TypeScript file. The `basePath` maybe be relative or absolute. All paths are resolved from the directory in which the minifier is executed. If there is no base path, the minified TypeScript file is outputted to the specified destination using a flattened file structure.
+
+`minifier.renameProgram()` takes a list of file names and an optional destination path:
+
+```javascript
+renameProgram(fileNames: string[], destination?: string)
+```
+
+`renameProgram` accepts TypeScript files and `.d.ts` files. If you are not explicitly including typings using the reference path syntax in your TypeScript file, please pass in the `.d.ts` file so that the minifier can use the type information.
+
+The minifier will uniformly rename the properties (which are available for renaming) across all the files that were passed in.
+
+###TSConfig Users
+
+---
+If you are using a `tsconfig.json` file to reference your type definitions, pass in the `.d.ts` files for libraries used in your program to the minifier so that it can reference the type information for property renaming. The minfier does not have access to the `tsconfig.json` file. Otherwise, make sure to have `/// <reference path = 'foo.d.ts' />` at the top of your programs if you are using any external libraries.
+
+###Scope of Minification
+
+---
+Currently, TS-Minify will do property renaming throughout all the files that are passed into it. The minifier excludes `.d.ts` files from renaming because those are the typing definitions of external libraries. If your TypeScript program exports objects across those files, the minifier will rename them uniformly in their declarations and their usage sites.
+
+There are plans to enable safer renaming if you intend for your TypeScript code to be used as an external library (in the case that you have a public API). At the moment, however, there is no support for this.
+
+###Caveats/Warnings
+
+---
+In order to get the most out of the minifier, it is recommended that the user types their program as specifically and thoroughly as possible. In general, you should  avoid using the `any` type, and implicit `any`s in your code as this might result in unwanted naming.  
+
+We recommend that you write TypeScript with the  “no implicit any” compiler flag enabled, and try transpiling your code with the TS compiler before trying the minifier.
```

Comment:
`write` -> `compile`. Write the flag in its commandline form `noImplicitAny` so people can copy paste it.